homeassistant/components/trafikverket_ferry/sensor.py

```diff
@@ -0,0 +1,140 @@
+
+
+@dataclass
+class TrafikverketSensorEntityDescription(
+    SensorEntityDescription, TrafikverketRequiredKeysMixin
+):
+    """Describes Trafikverket sensor entity."""
+
+
+SENSOR_TYPES: tuple[TrafikverketSensorEntityDescription, ...] = (
+    TrafikverketSensorEntityDescription(
+        key="departure_time",
+        name="Departure Time",
+        icon="mdi:clock",
+        device_class=SensorDeviceClass.TIMESTAMP,
+        value_fn=lambda data: data["departure_time"],
+        info_fn=lambda data: data["departure_information"],
+    ),
+    TrafikverketSensorEntityDescription(
+        key="departure_from",
+        name="Departure From",
+        icon="mdi:ferry",
+        value_fn=lambda data: data["departure_from"],
+        info_fn=lambda data: data["departure_information"],
+    ),
+    TrafikverketSensorEntityDescription(
+        key="departure_to",
+        name="Departure To",
+        icon="mdi:ferry",
+        value_fn=lambda data: data["departure_to"],
+        info_fn=lambda data: data["departure_information"],
+    ),
+    TrafikverketSensorEntityDescription(
+        key="departure_modified",
+        name="Departure Modified",
+        icon="mdi:clock",
+        device_class=SensorDeviceClass.TIMESTAMP,
+        value_fn=lambda data: data["departure_modified"],
+        info_fn=lambda data: data["departure_information"],
+        entity_registry_enabled_default=False,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the Trafikverket sensor entry."""
+
+    coordinator: TVDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+
+    async_add_entities(
+        [
+            FerrySensor(coordinator, entry.data[CONF_NAME], entry.entry_id, description)
+            for description in SENSOR_TYPES
+        ]
+    )
+
+
+class FerrySensor(CoordinatorEntity[TVDataUpdateCoordinator], SensorEntity):
+    """Contains data about a ferry departure."""
+
+    entity_description: TrafikverketSensorEntityDescription
+
+    def __init__(
+        self,
+        coordinator: TVDataUpdateCoordinator,
+        name: str,
+        entry_id: str,
+        entity_description: TrafikverketSensorEntityDescription,
+    ) -> None:
+        """Initialize the sensor."""
+        super().__init__(coordinator)
+        self._attr_name = f"{name} {entity_description.name}"
+        self.entity_description = entity_description
+        self._attr_device_info = DeviceInfo(
+            entry_type=DeviceEntryType.SERVICE,
+            identifiers={(DOMAIN, entry_id)},
+            manufacturer="Trafikverket",
+            model="v1.2",
+            name=name,
+            configuration_url="https://api.trafikinfo.trafikverket.se/",
+        )
+        self._attr_unique_id = f"{entry_id}-{entity_description.key}"
+
+    @property
+    def native_value(self) -> StateType:
+        """Return value of sensor."""
```

Comment:
You could implement _handle_coordinator_update and use the _attr shorthand

Reply:
I did like this
```python
    def _handle_coordinator_update(self) -> None:
        """Return state information."""
        self._attr_native_value = self.entity_description.value_fn(
            self.coordinator.data
        )
        self._attr_extra_state_attributes = {
            "other_information": self.entity_description.info_fn(self.coordinator.data),
        }
        return super()._handle_coordinator_update()
```
And it doesn't work. I tried several times before I never really understood how to use this

Reply:
Implement def _update_attr and call it from init and the handle coordinator update so state is set before the entity state is written to the state machine the first time

Reply:
Ok, got it.
But not sure I prefer the look off this code vs. the other.